src/filelock/_api.py

```diff
@@ -73,15 +75,36 @@ class ThreadLocalFileContext(FileLockContext, local):
     """A thread local version of the ``FileLockContext`` class."""
 
 
-class BaseFileLock(ABC, contextlib.ContextDecorator):
+class SingletonPerLockFileMeta(ABCMeta):
```

Comment:
We inherit from `ABCMeta` here rather than `type` since `BaseFileLock` inherits from `ABC` and we would otherwise have a metaclass conflict.